x-pack/plugins/security_solution/scripts/endpoint/common/endpoint_metadata_services.ts

```diff
@@ -131,3 +131,41 @@ const fetchLastStreamedEndpointUpdate = async (
 
   return queryResult.hits?.hits[0]?._source;
 };
+
+/**
+ * Waits for an endpoint to have streamed data to ES and for that data to have made it to the
+ * Endpoint Details API (transform destination index)
+ * @param kbnClient
+ * @param endpointAgentId
+ * @param timeoutMs
+ */
+export const waitForEndpointToStreamData = async (
+  kbnClient: KbnClient,
+  endpointAgentId: string,
+  timeoutMs: number = 60000
+): Promise<HostInfo> => {
+  const started = new Date();
+  const hasTimedOut = (): boolean => {
+    const elapsedTime = Date.now() - started.getTime();
+    return elapsedTime > timeoutMs;
+  };
+  let found: HostInfo | undefined;
+
+  while (!found && !hasTimedOut()) {
+    found = await fetchEndpointMetadata(kbnClient, endpointAgentId).catch((error) => {
+      // FIXME:PT check errors, ignore 404
```

Comment:
Not sure if this is for a later PR

Reply:
ahhh.. yes, I should implement this. Goal is to ignore `404` if the `agentId` is not found. I'll implement it now and will push another commit